### PR TITLE
Dotnet8rc

### PR DESCRIPTION
**Pull request by Xerxes-2**
handle rc version, update preview to newest rc

### CODE REVIEW

File: auto_updater.py [hunk 1]
```diff
@@ -142,12 +151,21 @@ def __init__(self, version_string):
...
-
+        self.version_tuple = (
+            self.major,
+            self.minor,
+            self.patch,
+            self.is_rc,
+            self.update,
+            self.build,
+            self.update_specific_build
+        )
```
Comment (isen-ng) [inline auto_updater.py:168]: New line here to adhere to python style

File: auto_updater.py [hunk 2]
```diff
@@ -98,16 +98,27 @@ def __lt__(self, other):
...
         self.minor = int(version_split[1])
         self.feature = int(version_split[2][0])
-        self.patch = int(version_split[2][1:-8])
+        self.patch = int(version_split[2].split('-')[0][1:])
+        self.is_rc = 'rc' in version_split[2]
         self.update = int(version_split[3])
         self.build = int(version_split[4])
         self.update_specific_build = int(version_split[5])
         self.version_string = version_string
+        self.version_tuple = (
```
Comment (isen-ng) [inline auto_updater.py:112]: I'm not very good in python, so this is my first time encountering a python tuple. It's very convenient!

But as a result, there is no longer a need to have the other `self.` variables right? Only `self.version_string` and `self.version_tuple` is needed. Could I trouble you to clean them up?

File: auto_updater.py [hunk 3]
```diff
@@ -142,12 +151,21 @@ def __init__(self, version_string):
...
         self.minor = int(version_split[1])
-        self.patch = int(version_split[2][:-8])
+        self.patch = int(version_split[2].split('-')[0])
+        self.is_rc = 'rc' in version_split[2]
         self.update = int(version_split[3])
         self.build = int(version_split[4])
         self.update_specific_build = int(version_split[5])
         self.version_string = version_string
-
+        self.version_tuple = (
```
Comment (isen-ng) [inline auto_updater.py:160]: Same here about the redundant `self ` variables

**Review by isen-ng — CHANGES_REQUESTED** [review]

**Comment by isen-ng** [comment]
Thank you @Xerxes-2 for the PR as well as the update to the updater. I wouldn't have caught it otherwise.

On the other hand, could I also trouble you to update the preview versions in the README.md file?

https://github.com/isen-ng/homebrew-dotnet-sdk-versions/blob/master/README.md#preview-versions

**Comment by Xerxes-2** [comment]
Thank you for review @isen-ng ! I have resolved these issues and updated README.

**Review by isen-ng — APPROVED** [review]

**Comment by isen-ng** [comment]
Thank you for your changes! LGTM

**Pull request merged**